### ISSUE TITLE
Descriptive error reporting from maap py

### ISSUE THREAD

**Issue by sujen1412** [MAAP-Project/ZenHub#906]
Currently if something fails on the api when using maap-py to submit job or register algorithm, the error returned is not useful. 
Eg. Error 500 instead of Algorithm not found. 

Update error reporting to be more descriptive for the user.

**Comment by chuckwondo** [comment]
To be more specific, this was discovered when calling the `submitJob` in the maap-py lib with a value for `algo_id` for an algorithm that doesn't exist. In this case, the MAAP API seems to simply not ensure that the supplied `algo_id` is actually the name of a registered algorithm, resulting in the following being returned from `submitJob`:

```
{'job_id': '', 'status': 'failed', 'machine_type': None, 'architecture': None, 'machine_memory_size': None, 'directory_size': None, 'operating_system': None, 'job_start_time': None, 'job_end_time': None, 'job_duration_seconds': None, 'cpu_usage': None, 'cache_usage': None, 'mem_usage': None, 'max_mem_usage': None, 'swap_usage': None, 'read_io_stats': None, 'write_io_stats': None, 'sync_io_stats': None, 'async_io_stats': None, 'total_io_stats': None, 'error_details': '<ows:ExceptionReport xmlns:wps="http://www.opengis.net/wps/2.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:schemaLocation="http://schemas.opengis.net/wps/2.0/wps.xsd" xmlns:ows="http://www.opengis.net/ows/2.0"><ows:Exception exceptionCode="FailedJobSubmit" locator="Execute"><ows:ExceptionText>Failed to submit job of type job-gedi-subset_ubuntu:0.6.2. Exception Message: \'NoneType\' object has no attribute \'get\'</ows:ExceptionText></ows:Exception></ows:ExceptionReport>', 'response_code': 500, 'outputs': []}
```

So this is not necessarily about "improving" error reporting, but about performing a "null" check to avoid throwing a Python error: `'NoneType' object has no attribute 'get'`

In this case, the response should be a 404 (not 500, due to the unhandled exception), with a message similar to `Algorithm not found: ALGO_ID`, where `ALGO_ID` is the value of the `algo_id` parameter passed into the `submitJob` function.

**Comment by sujen1412** [comment]
Yes is this particular case it was for algorithm not existing. In addition to the above, I have seen it to cause issues when incorrect parameters are specified as well. So hoping to cover more such cases in this ticket.
